src/ws/ssh.c

```diff
@@ -1207,23 +1208,32 @@ cockpit_ssh_connect (CockpitSshData *data,
       goto out;
     }
 
+  /* if we check the default system known hosts file (i. e. not during the test
+   * suite), also check the legacy file in /var/lib/cockpit; we need to do that
+   * even with allow_unknown_hosts as subsequent code relies on knownhosts_file */
+  host_known = cockpit_is_host_known (data->ssh_options->knownhosts_file, host, port);
+  if (!host_known && strncmp (data->ssh_options->knownhosts_file, PACKAGE_SYSCONF_DIR, strlen (PACKAGE_SYSCONF_DIR)) == 0)
+    {
+      g_debug ("%s: %s is not known in %s, falling back to %s",
+              data->logname,
+              data->ssh_options->knownhosts_file, PACKAGE_LOCALSTATE_DIR "/known_hosts");
+      data->ssh_options->knownhosts_file = PACKAGE_LOCALSTATE_DIR "/known_hosts";
+      host_known = cockpit_is_host_known (data->ssh_options->knownhosts_file, host, port);
```

Comment:
Hmm, i think it might actually be preferrable to compare `data->ssh_options->knownhosts` to the complete default path, not just `/etc`. I. e. as soon as the caller specifies something custom we don't want the fallback.